cime_config/hist_config.py

```diff
@@ -161,52 +161,50 @@ def _is_mult_period(entry):
     >>> _is_mult_period("3 * nmonths")
     ((3, 'nmonths'), None)
     >>> _is_mult_period("2*fortnights")
-    (None, 'period must be one of nsteps, nstep, nseconds, nsecond, nminutes, nminute, nhours, nhour, ndays, nday, nmonths, nmonth, nyears, nyear, steps, seconds, minutes, hours, days, months, years')
+    (None, 'period ("fortnights") must be one of nsteps, nstep, nseconds, nsecond, nminutes, nminute, nhours, nhour, ndays, nday, nmonths, nmonth, nyears, nyear, steps, seconds, minutes, hours, days, months, years')
     >>> _is_mult_period("7*nhours of day")
-    (None, 'period must be one of nsteps, nstep, nseconds, nsecond, nminutes, nminute, nhours, nhour, ndays, nday, nmonths, nmonth, nyears, nyear, steps, seconds, minutes, hours, days, months, years')
+    (None, 'period ("nhours of day") must be one of nsteps, nstep, nseconds, nsecond, nminutes, nminute, nhours, nhour, ndays, nday, nmonths, nmonth, nyears, nyear, steps, seconds, minutes, hours, days, months, years')
     >>> _is_mult_period(" ")
-    (None, 'a frequency ([<mult>*]period) is required')
+    (None, 'frequency ([<mult>*]period) is required, found " "')
     >>> _is_mult_period("1*nyear")
     ((1, 'nyear'), None)
     >>> _is_mult_period("-6*nhours")
-    (None, "multiplier '-6' in '-6*nhours' must be a positive integer")
+    (None, 'multiplier ("-6") must be a positive integer')
     >>> _is_mult_period("7h*nhours")
-    (None, "multiplier '7h' in '7h*nhours' must be a positive integer")
+    (None, '"7h" in "7h*nhours" is not a valid integer')
     >>> _is_mult_period("5*nhours*ndays")
-    (None, "multiplier '5*nhours' in '5*nhours*ndays' must be a positive integer")
+    (None, 'frequency must be of the form ([<mult>*]period), found "5*nhours*ndays".  Do you have too many multipliers or periods?')
     """
-    if not entry.strip():
-        errmsg = "a frequency ([<mult>*]period) is required"
-        return (None, errmsg)
+    if not entry or not entry.strip():
+        return (None, f"frequency ([<mult>*]period) is required, found \"{entry}\"")
     # end if
-    _OPTIONAL_MULTIPLIER_TOKEN = r"((?P<multiplier>.*)\*)"
-    _REQUIRED_PERIOD_TOKEN = "(?P<period>.*)"
-    _MULT_PERIOD_REGEX = f"^{_OPTIONAL_MULTIPLIER_TOKEN}?\s*{_REQUIRED_PERIOD_TOKEN}$"
-    match = re.match(_MULT_PERIOD_REGEX, entry.strip())
-    group_dict = match.groupdict()
-    if len(group_dict) == 0 or 'period' not in group_dict:
-        errmsg = f"Bad formatting of frequency, '{entry}' must be in the form of '[<integer>*]<time period>'"
-        return (None, errmsg)
-    # end if
-    # Check multiplier
+    tokens = [x.strip() for x in str(entry.strip()).split('*')]
```

Comment:
I think `entry` is already a string, so no need to convert it here.  Also I don't think there is any need to `strip` twice:

```suggestion
    tokens = [x.strip() for x in entry.split('*')]
```